packages/app/src/game/components/field-cell/field-cell.tsx

```diff
@@ -69,27 +99,22 @@ const FieldCellComponent = (props: Props) => {
     // eslint-disable-next-line no-undefined
     const handlePress = () => void onSelect(isActive ? undefined : cell);
 
-    const cellStyles = [
-        styles.container,
-        cs(sudoku.isLastInCellGroupX(cell), styles.groupXEnd),
-        cs(sudoku.isLastInCellGroupY(cell), styles.groupYEnd),
-        cs(isLastRow, styles.lastRow),
-        cs(isLastCol, styles.lastCol),
-        { backgroundColor },
-        animatedStyles
-    ];
+    const cellStyles = getCellStyles(sudoku, cell, backgroundColor, animatedStyles);
 
     return (
-        <ReanimatedPressable onPress={handlePress} style={cellStyles} testID={getCellSelector(props)}>
+        <ReanimatedPressable 
+            onPress={handlePress} 
+            style={cellStyles} 
+            testID={getCellSelector(props)}
+        >
             <FieldCellText
```

Comment:
Let's inline cell text here and remove the component? Also merge styles as this will gather the logic and avoid props strolling